src/pages/ProfilesExplorerView/components/SceneLabelValuesTimeseries.tsx

```diff
@@ -310,9 +299,110 @@ export class SceneLabelValuesTimeseries extends SceneObjectBase<SceneLabelValues
+    const titleParts = [parsedQuery.serviceId, getProfileMetricLabel(parsedQuery.profileMetricId)];
+
+    if (groupBy?.length) {
+      titleParts.push(groupBy[0]);
+    }
+
+    if (parsedQuery.labels.length) {
+      titleParts.push(parsedQuery.labels.join(', '));
+    }
+
+    const title = titleParts.join(' · ');
+    const datasource = sceneGraph.interpolate(this, '${dataSource}');
+    const timeRange = sceneGraph.getTimeRange(this).state.value;
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    return useMemo(() => {
+      return {
+        id: nanoid(),
+        origin: 'Explore Profiles',
+        url: window.location.href,
+        logoPath: PyroscopeLogo,
+        title,
+        type: 'timeseries',
+        timeRange: { ...timeRange },
+        queries: [{ refId, queryType, profileTypeId, labelSelector, groupBy }],
+        datasource,
+      };
+    }, [datasource, groupBy, labelSelector, profileTypeId, queryType, refId, timeRange, title]);
+  }
+
+  buildMenuItems(selectedScaleType: ScaleDistribution, addToInvestigationLink?: PluginExtensionLink) {
+    const menuItems: PanelMenuItem[] = [
+      {
+        text: 'Scale type',
+        type: 'group',
+        subMenu: [
+          {
+            text: 'Linear',
+            scaleDistribution: { type: ScaleDistribution.Linear },
+          },
+          {
+            text: 'Log2',
+            scaleDistribution: { type: ScaleDistribution.Log, log: 2 },
+          },
+        ].map((option) => ({
+          text: `${selectedScaleType === option.scaleDistribution.type ? '✔ ' : ''}${option.text}`,
+          onClick: () => this.onClickScaleOption(option),
+        })),
+      },
+      {
+        type: 'divider',
+        text: '',
+      },
+      {
+        iconClassName: 'compass',
+        text: 'Open in Explore',
+        onClick: () => this.onClickExplore(),
+      },
+    ];
+
+    if (addToInvestigationLink) {
+      menuItems.push({
+        iconClassName: 'plus-square',
+        text: 'Add to investigation',
+        onClick: () => {
+          addToInvestigationLink.onClick!();
+        },
+      });
+    }
+
+    return menuItems;
+  }
+
+  useUpdateMenuItems() {
+    const { body, scaleType } = this.state;
+    const { menu } = body.state;
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const context = this.useGetInvestigationPluginLinkContext();
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const link = useGetPluginExtensionLink({
+      extensionPointId: INVESTIGATIONS_EXTENSTION_POINT_ID,
+      context,
+      pluginId: INVESTIGATIONS_APP_ID,
+    });
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useEffect(() => {
+      // wrapped in a useEffect to prevent issues when clicking on the "add to investigation" link
+      // ("Cannot update a component while rendering a different component")
+      menu?.setItems(this.buildMenuItems(scaleType, link));
```

Comment:
This feels weird because:
- on one side: `onClickScaleOption` sets the new value for `scaleType` but does not update the menu items
- on the other side: every time the link changes (side-effect) or the `scaleType` changes, the menu items are updated

I think I should modify the code so that:
- `onClickScaleOption` updates only its menu items
- whenever the link is available or changes, we modify the "Add to investigation"  link only

Wdyt?